### DISCUSSION TITLE
Vriting FFMpeg video in an extension.

### DISCUSSION THREAD

**Discussion by vigji** [bonsai-rx/bonsai#1732] [Q&A] [answered]
Hi, I have the following simple schema for data acquisition from two cameras:

![main](https://github.com/bonsai-rx/bonsai/assets/10400212/840941f2-2526-40d1-b476-6198db72776b)

Where the content of the TimestampedVideoWriter extension is the following:

![videosave](https://github.com/bonsai-rx/bonsai/assets/10400212/22549b6d-687a-4724-9c2c-2391e4bef318)

Don't know if I am making sense, but I wanted to organize that in an extension as I am using the pattern many times in different subprojects.

However, it seems that to be working I cannot use the same Extension for two cameras: I have to duplicate the Extension `.bonsai` file to two identical files, otherwise I get funny errors that make me think that there is some overlapping calls to the same nodes (filename already existing, pipeline busy...). What am I missing with this pattern? I thought that each usage of a Workflow saved as an Extension would be independent from the others.

Here is the zipped code:

[testing_extensions.zip](https://github.com/bonsai-rx/bonsai/files/14943637/testing_extensions.zip)

**Comment by bruno-f-cruz** [comment, marked as answer]
I have attached an example that i think does what you are trying to achieve.

[TimestampedVideoWriter_Refactored.zip](https://github.com/bonsai-rx/bonsai/files/14947609/TimestampedVideoWriter_Refactored.zip)

On a side note, be very careful when using the timestamp operator to align video data, the jitter is quite large. You should instead rely on hardware-level synchronization as discussed in this repository several times before.

**Comment by vigji** [comment]
This works like a charm, and also super helpful to learn more patterns!

Thanks for the tip, yes, the cameras are hardware-level snched with TTLs, but as they're started at every trial I wanted to have this logging as a sanity check.

Thanks a lot and sorry for the late testing!
